src/main/java/com/google/sps/servlets/NewRecipeServlet.java

```diff
@@ -37,30 +38,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // redirects user back to UserPage
     // TODO: inform the user that they're missing stuff
     if (name.equals("") || tagsResponse.equals("") || description.equals("") || ingredientsResponse.equals("") || stepsResponse.equals("")){
```

Comment:
I know this isn't technically part of this PR, but you probably also want to trim the responses before checking that they're empty here, ie: 
```
if (name.trim().isEmpty() || tagsResponse.trim().isEmpty() .... )